src/main/java/com/google/research/bleth/simulator/AbstractSimulation.java

```diff
@@ -94,10 +106,35 @@ void resolverEstimate() {
     void writeRoundState() { }
 
     /** Gather statistical data of the current round and update the aggregated simulation statistics based on all rounds. */
-    void updateSimulationStats() { }
+    void updateSimulationStats() {
+        Map<Beacon, Location> beaconsToEstimatedLocations = ((GlobalResolver) resolver).getBeaconsToEstimatedLocations();
+        List<Double> distances = beaconsToEstimatedLocations.keySet().stream() // ignore beacons that have never been observed
+                .map(beacon -> distance(beaconsToEstimatedLocations.get(beacon), beacon.getLocation())).collect(toList());
+
+        if (!distances.isEmpty()) { // distances is empty until the first round an observer observed a beacon
+            double min = distances.stream().min(Double::compareTo).get();
+            distancesStats.put("min", Math.min(distancesStats.getOrDefault("min", Double.POSITIVE_INFINITY), min));
+
+            double max = distances.stream().max(Double::compareTo).get();
+            distancesStats.put("max", Math.max(distancesStats.getOrDefault("max", Double.NEGATIVE_INFINITY), max));
+
+            double average = distances.stream().mapToDouble(Double::doubleValue).sum() / distances.size();
+            double allRoundsAverage = (distancesStats.getOrDefault("avg", 0D) * (currentRound - 1) + average) / currentRound;
+            distancesStats.put("avg", allRoundsAverage);
+        }
+    }
 
     /** Write final simulation statistical data to db. */
-    void writeSimulationStats() { }
+    void writeSimulationStats() {
+        for (Beacon beacon : beacons) {
+            String beaconId = String.valueOf(beacon.getId());
+            beaconsObservedSum.putIfAbsent(beaconId, 0D);
+            beaconsObservedSum.put(beaconId, beaconsObservedSum.get(beaconId) / (currentRound - 1));
```

Comment:
Why are you dividing by the round number?
(This actually does deserve a comment...)